token/js/test/e2e-2022/tokenMetadata.test.ts

```diff
@@ -0,0 +1,421 @@
+    });
+});
+
+describe('Token Metadata operations', () => {
+    const EXTENSIONS = [ExtensionType.MetadataPointer];
+    const mintLen = getMintLen(EXTENSIONS);
+
+    let connection: Connection;
+    let payer: Signer;
+    let mint: Keypair;
+    const authority = Keypair.fromSecretKey(
+        new Uint8Array([
+            118, 177, 37, 231, 15, 88, 210, 92, 79, 231, 202, 22, 11, 15, 121, 54, 95, 229, 149, 119, 48, 177, 187, 198,
+            223, 51, 225, 74, 12, 54, 172, 36, 207, 107, 122, 208, 209, 168, 61, 177, 190, 137, 23, 156, 84, 32, 34, 82,
+            158, 176, 55, 51, 236, 66, 130, 167, 118, 31, 120, 107, 100, 192, 147, 10,
+        ])
+    );
+
+    before(async () => {
+        connection = await getConnection();
+        payer = await newAccountWithLamports(connection, 1000000000);
+    });
+
+    beforeEach(async () => {
+        mint = Keypair.generate();
+
+        const lamports = await connection.getMinimumBalanceForRentExemption(mintLen);
+
+        const transaction = new Transaction().add(
+            SystemProgram.createAccount({
+                fromPubkey: payer.publicKey,
+                newAccountPubkey: mint.publicKey,
+                space: mintLen,
+                lamports: lamports,
+                programId: TEST_PROGRAM_ID,
+            }),
+            createInitializeMetadataPointerInstruction(
+                mint.publicKey,
+                authority.publicKey,
+                mint.publicKey,
+                TEST_PROGRAM_ID
+            ),
+            createInitializeMintInstruction(
+                mint.publicKey,
+                TEST_TOKEN_DECIMALS,
+                authority.publicKey,
+                null,
+                TEST_PROGRAM_ID
+            )
+        );
+
+        await sendAndConfirmTransaction(connection, transaction, [payer, mint], undefined);
+
+        await tokenMetadataInitializeWithRentTransfer(
+            connection,
+            payer,
+            authority.publicKey,
+            mint.publicKey,
+            authority,
+            'name',
+            'symbol',
+            'uri',
+            undefined,
+            undefined,
+            TEST_PROGRAM_ID
+        );
+    });
+
+    it('can successfully initialize', async () => {
+        const meta = await getTokenMetadata(connection, mint.publicKey, undefined, TEST_PROGRAM_ID);
+        expect(meta).to.deep.equal({
+            updateAuthority: new PublicKey('ExgT3gCWXJzY4a9SHqTqsTk6dPAj37WNq2uWNbmMG1JR'),
+            mint: mint.publicKey,
+            name: 'name',
+            symbol: 'symbol',
+            uri: 'uri',
+            additionalMetadata: [],
+        });
+    });
+
+    it('can successfully emit', async () => {
+        const signature = await tokenMetadataEmit(connection, payer, mint.publicKey);
+
+        // Get metadata emitted in the transaction to verify
+        const tx: any = await connection.getTransaction(signature, {
```

Comment:
I removed the `getEmittedTokenMetadata` but kept the logic here just to check the emit worked. 

Can remove if you want - feels more like i'm testing the token program rather than the js client